spring-kafka-test/src/main/java/org/springframework/kafka/test/rule/KafkaEmbedded.java

```diff
@@ -137,10 +139,12 @@ public void before() throws Exception { //NOSONAR
 		this.zkConnect = "127.0.0.1:" + this.zookeeper.port();
 		this.zookeeperClient = new ZkClient(this.zkConnect, zkSessionTimeout, zkConnectionTimeout,
 				ZKStringSerializer$.MODULE$);
-		this.kafkaServers = new ArrayList<>();
+		this.kafkaServers.clear();
+		this.ports.clear();
 		for (int i = 0; i < this.count; i++) {
 			ServerSocket ss = ServerSocketFactory.getDefault().createServerSocket(0);
 			int randomPort = ss.getLocalPort();
+			this.ports.add(randomPort);
```

Comment:
Looks like not a big deal to populate the `port` into config overriding that default one: https://github.com/artembilan/spring-kafka/commit/65397705c046fcdff76e5ed014b20d2b74f8e3e1

Let me know and I'll merge with that fix